eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/ReplaceableTxFunder.scala

```diff
@@ -405,75 +353,46 @@ private class ReplaceableTxFunder(nodeParams: NodeParams,
-    // Thus we have: anchorFeerate = targetFeerate + (weight-commit-tx / weight-anchor-tx) * (targetFeerate - commitTxFeerate)
-    // If we use the smallest weight possible for the anchor tx, the feerate we use will thus be greater than what we want,
-    // and we can adjust it afterwards by raising the change output amount.
-    val anchorFeerate = targetFeerate + FeeratePerKw(targetFeerate.feerate - commitFeerate.feerate) * commitTx.weight() / claimAnchorOutputMinWeight
     // NB: fundrawtransaction requires at least one output, and may add at most one additional change output.
     // Since the purpose of this transaction is just to do a CPFP, the resulting tx should have a single change output
-    // (note that bitcoind doesn't let us publish a transaction with no outputs).
-    // To work around these limitations, we start with a dummy output and later merge that dummy output with the optional
-    // change output added by bitcoind.
-    // NB: fundrawtransaction doesn't support non-wallet inputs, so we have to remove our anchor input and re-add it later.
-    // That means bitcoind will not take our anchor input's weight into account when adding inputs to set the fee.
-    // That's ok, we can increase the fee later by decreasing the output amount. But we need to ensure we'll have enough
-    // to cover the weight of our anchor input, which is why we set it to the following value.
-    val dummyChangeAmount = weight2fee(anchorFeerate, claimAnchorOutputMinWeight) + dustLimit
-    val txNotFunded = Transaction(2, Nil, TxOut(dummyChangeAmount, Script.pay2wpkh(PlaceHolderPubKey)) :: Nil, 0)
-    bitcoinClient.fundTransaction(txNotFunded, FundTransactionOptions(anchorFeerate, lockUtxos = true)).flatMap(fundTxResponse => {
+    // (note that bitcoind doesn't let us publish a transaction with no outputs). To work around these limitations, we
+    // start with a dummy output and later merge that dummy output with the optional change output added by bitcoind.
+    val txNotFunded = anchorTx.txInfo.tx.copy(txOut = TxOut(dustLimit, Script.pay2wpkh(PlaceHolderPubKey)) :: Nil)
+    // The anchor transaction is paying for the weight of the commitment transaction.
+    val anchorWeight = Seq(InputWeight(anchorTx.txInfo.input.outPoint, anchorInputWeight + commitTx.weight()))
+    bitcoinClient.fundTransaction(txNotFunded, FundTransactionOptions(targetFeerate, lockUtxos = true, inputWeights = anchorWeight)).flatMap(fundTxResponse => {
       // We merge the outputs if there's more than one.
       fundTxResponse.changePosition match {
         case Some(changePos) =>
           val changeOutput = fundTxResponse.tx.txOut(changePos)
-          val txSingleOutput = fundTxResponse.tx.copy(txOut = Seq(changeOutput.copy(amount = changeOutput.amount + dummyChangeAmount)))
-          Future.successful(fundTxResponse.copy(tx = txSingleOutput))
+          val txSingleOutput = fundTxResponse.tx.copy(txOut = Seq(changeOutput))
```

Comment:
Do we still need to make this a single output tx ?

Reply:
Yes, it wouldn't make sense to have two change outputs, right (and it would cost more in terms of weight)? Since this tx is only used to do CPFP, its only output will be change. Ideally `bitcoind` would let us call `fundrawtransaction` without any output so we wouldn't need this "hack", but they don't because transactions without outputs aren't consensus valid...